### PR TITLE
docs: Add USAGE.md

### LINKED ISSUE #44: Copy docs from a PR to create a USAGE.md file
#### Issue Summary

Usage files are really nice when they exist in a repo, because it makes it really obvious where to look for information about how to use this tool. It can be frustrating when you expect this file to be there and it's not. 
Please add a USAGE.md file, you can see an example of this 
[here](https://github.com/sendgrid/sendgrid-python/blob/master/USAGE.md)

Please make sure you modify the file for this repo (hint: Most of this information should be in the README file)

Also, @fifiteen82726 [put together some AWESOME docs](https://github.com/sendgrid/ruby-http-client/pull/43) for how to install and run codeclimate locally, including a gif. Please add this to your new USAGE.md file.

### PR DESCRIPTION
**Pull request by ProZsolt**
# Fixes #44

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Usage
+
+Usage examples for SendGrid ruby-http-client
+
+## Initialization
+
+```
```
Comment (kplimack) [inline USAGE.md]: add `ruby` code formatter
i.e.
```
 ```ruby
 <code>
 ....
```

File: USAGE.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+')
+response = client.version('v3').api_keys.post(request_body: request_body)
+puts response.status_code
+puts response.body
+puts response.headers
+api_key_id = JSON.parse(response.body)['api_key_id']
+```
+
+<a name="put"></a>
```
Comment (kplimack) [inline USAGE.md]: markdown automatically generates anchors to each header, you can remove all the html tags
`[create an anchor](#anchors-in-markdown)`

File: USAGE.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        ]
+    }
+')
+response = client.api_keys._(api_key_id).put(request_body: request_body)
+puts response.status_code
+puts response.body
+puts response.headers
+```
+
+<a name="patch"></a>
```
Comment (kplimack) [inline USAGE.md]: markdown automatically generates anchors to each header, you can remove all the html tags

File: USAGE.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+puts response.status_code
+puts response.body
+puts response.headers
+api_key_id = JSON.parse(response.body)['api_key_id']
+```
+
+<a name="put"></a>
+## PUT
+
+```
```
Comment (kplimack) [inline USAGE.md]: add ruby formatter

File: USAGE.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+<a name="delete"></a>
+## DELETE
+
+```
+response = client.api_keys._(api_key_id).delete
+puts response.status_code
+puts response.headers
+```
+
+<a name="post"></a>
```
Comment (kplimack) [inline USAGE.md]: markdown automatically generates anchors to each header, you can remove all the html tags

File: USAGE.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+response = client.version('v3').api_keys._(api_key_id).get
+puts response.status_code
+puts response.body
+puts response.headers
+```
+
+<a name="delete"></a>
+## DELETE
+
+```
```
Comment (kplimack) [inline USAGE.md]: ruby formatter

File: USAGE.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+query_params = { 'limit' => 100, 'offset' => 0 }
+response = client.version('v3').api_keys.get(query_params: query_params)
+puts response.status_code
+puts response.body
+puts response.headers
+```
+
+#### GET Single
+
+```
```
Comment (kplimack) [inline USAGE.md]: ruby formatter

File: USAGE.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+response = client.version('v3').api_keys._(api_key_id).get
+puts response.status_code
+puts response.body
+puts response.headers
+```
+
+<a name="delete"></a>
```
Comment (kplimack) [inline USAGE.md]: markdown automatically generates anchors to each header, you can remove all the html tags

File: USAGE.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+## Table of Contents
+
+- [GET](#get)
+- [DELETE](#delete)
+- [POST](#post)
+- [PUT](#put)
+- [PATCH](#patch)
+
+<a name="get"></a>
```
Comment (kplimack) [inline USAGE.md]: markdown automatically generates anchors to each header, you can remove all the html tags

File: USAGE.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+## GET
+
+#### GET Collection
+
+```
```
Comment (kplimack) [inline USAGE.md]: ruby formatter

File: USAGE.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+response = client.api_keys._(api_key_id).delete
+puts response.status_code
+puts response.headers
+```
+
+<a name="post"></a>
+## POST
+
+```
```
Comment (kplimack) [inline USAGE.md]: ruby formatter

File: USAGE.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+response = client.api_keys._(api_key_id).put(request_body: request_body)
+puts response.status_code
+puts response.body
+puts response.headers
+```
+
+<a name="patch"></a>
+## PATCH
+
+```
```
Comment (kplimack) [inline USAGE.md]: ruby formatter

**Review by kplimack — CHANGES_REQUESTED** [review]
@ProZsolt thank you for your contribution!  I left a couple small requests for minor updates.

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/ruby-http-client?pullRequest=72) <br/>All committers have signed the CLA.

**Review by kplimack — APPROVED** [review]

**Comment by thinkingserious** [comment]
Hello @ProZsolt,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @ProZsolt,<br /><br/>Thanks again for the PR!<br /><br />                                         We noticed that you have earned 3 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold.<br /><br />                                         Also, please be sure that you have officially registered <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @ProZsolt,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 3 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid

**Comment by ProZsolt** [comment]
Sadly, I will miss out on this :( I would be happy to contribute more, but
I'm currently on a road trip around New Zealand. Where I barely have
internet or electricity. Last year it was one merged PR. This year it
wasn't mentioned it changed, that's why I did my PRs in the begining of
October. I also got confirmation when my other two PRs was merged.
Please next time make sure if something changes let people know immediately
and not it the end when they only have few days to react.
It's a bit of let down, but see you guys next year.

**Pull request merged**